locksmith/src/websub/jobs/keys.ts

```diff
@@ -1,19 +1,42 @@
 import { Key } from '../../graphql/datasource'
-import { Hook } from '../../models'
+import { Hook, ProcessedHookItem } from '../../models'
 import { TOPIC_KEYS } from '../topics'
 
 import { networkMapToFnResult, notifyHook } from '../helpers'
 
+async function getUnprocessedKeys(network: number) {
+  const keySource = new Key(network)
+  const processedKeys = await ProcessedHookItem.findAll({
```

Comment:
I thought of limiting this to not strain the database. What do you think of a limit of getting the last 100 items which fullfil this?

Reply:
So I would do it in reverse. Rather than get object that have been processed, I would get a list of object "toProcess" from the subgraph and then look which of them have already been processed and only keep the ones which have not. Additionnaly if _all of them_ have not been processed, than it means we can use the subgraph to retrieve more keys and try to process them.

What do you think?